pkg/data/management/catalog_data.go

```diff
@@ -193,10 +198,17 @@ func doAddCatalogs(management *config.ManagementContext, name, url, branch, helm
 				HelmVersion: helmVersion,
 			},
 		}
-		if _, err := catalogClient.Create(obj); err != nil {
+		if obj, err = catalogClient.Create(obj); err != nil {
 			return err
 		}
 	}
 
+	if bundledMode && obj.Name == utils.SystemLibraryName {
+		// force update the catalog cache on every startup
+		configMapInterface := management.Core.ConfigMaps("")
+		configMapLister := configMapInterface.Controller().Lister()
+		return manager.CreateOrUpdateSystemCatalogImageCache(obj, configMapInterface, configMapLister, true, true)
```

Comment:
For this invocation of `CreateOrUpdateSystemCatalogImageCache`, I am specifying that we always do a forceUpdate since, if a user were to change the contents of `/var/lib/rancher-data/local-catalogs/system-library`, they should be able to update the contents of the `/v3/kontainerdriver/rancher-images` simply by restarting Rancher. 

Without this, there would be no way of supporting upgrading the system-charts and seeing the new images pop up in air-gapped setups.

Reply:
This context should also be added as a comment on L207 to explain why we're force updating the catalog cache.

Reply:
PS: how does `system-library` content affect `kontainerdriver` images? we only cache images from chart in `AddImagesToImageListConfigMap`?

Reply:
Sure, I will add it as a comment.

Reply:
> PS: how does system-library content affect kontainerdriver images? we only cache images from chart in AddImagesToImageListConfigMap?

`kontainerdriver` images are composed of images from 3 sources, as shown in https://github.com/rancher/rancher/blob/fa481ddb14e3a3413ca5d75e709474bfb02a7eab/pkg/api/norman/customization/kontainerdriver/actionhandler.go#L183-L185

1. The agent image
2. RKESystemImages (from KDM)
3. The system-chart images

Prior to 2.6, the contents of the git repo for system-charts in the `management-state` directory on each Rancher container was the source of truth for the latest copy of system-charts. We would `GetImages` on demand when such a request came in.

However, to fix #31336, we made the new source of truth for system-chart images the `system-library` ConfigMap instead, which already contains the output of the `GetImages` command on the system-charts.

Therefore, `AddImagesToImageListConfigMap` gets called in one of two ways:
1. On startup (after my fix) to create the initial ConfigMap and update it based on the bundled contents
2. On update (only in non-bundled Rancher clusters), as a result of the catalog `manager.Sync` if it is detected that the catalog is not up-to-date.